### PR TITLE
wip staticarrays

### PR DESCRIPTION
**Pull request by KristofferC**
Also adds tests for Float32.

Benchmarks: https://gist.github.com/KristofferC/fe50195e9beba7b3d3e252b359a06521

Need to look at: `otimes`, `skew`, `dev`

### CODE REVIEW

**Comment by KristofferC** [comment]
EEh lol.. Might have compared the wrong way.

File: src/ContMechTensors.jl [hunk 1]
```diff
@@ -74,31 +88,31 @@ function n_independent_components(dim, issym)
...
     n = n_components(SymmetricTensor{2, dim})
     return n*n
 end
 
-@inline n_components{order, dim}(::Type{Tensor{order, dim}}) = dim^order
+@pure n_components{order, dim}(::Type{Tensor{order, dim}}) = dim^order
 
-@inline is_always_sym{dim, T}(::Type{Tensor{dim, T}}) = false
-@inline is_always_sym{dim, T}(::Type{SymmetricTensor{dim, T}}) = true
+@pure is_always_sym{dim, T}(::Type{Tensor{dim, T}}) = false
```
Comment (fredrikekre) [inline src/ContMechTensors.jl]: These are never used btw.

**Comment by fredrikekre** [comment]
Is there a new compare if the one you linked is wrong?

**Comment by KristofferC** [comment]
https://gist.github.com/KristofferC/949adf01633dee6a6837a7520de472f5

**Comment by fredrikekre** [comment]
Some symmetric operations becomes much slower?

**Comment by fredrikekre** [comment]
| `["otimes","dim 2 - order 2sym 2sym - Float32"]` | 3.35 (5%) :x: | 1.00 (1%)  |
| `["otimes","dim 2 - order 2sym 2sym - Float64"]` | 3.16 (5%) :x: | 1.00 (1%)  |

| `["otimes","dim 3 - order 2sym 2sym - Float32"]` | 2.18 (5%) :x: | 1.00 (1%)  |
| `["otimes","dim 3 - order 2sym 2sym - Float64"]` | 2.55 (5%) :x: | 1.00 (1%)  |

File: src/ContMechTensors.jl [hunk 2]
```diff
@@ -2,6 +2,11 @@ __precompile__()
 
 module ContMechTensors
 
+import Base.@pure
+
+using StaticArrays
+import StaticArrays.setindex
```
Comment (fredrikekre) [inline src/ContMechTensors.jl:8]: This is not in the tagged version of `StaticArrays`. Need to add the new tag (when its tagged) to `REQUIRE` later

Comment (KristofferC) [reply src/ContMechTensors.jl:8]: Yep

Comment (fredrikekre) [reply src/ContMechTensors.jl:8]: If we even have to support it? I have never used it I think :)

Comment (KristofferC) [reply src/ContMechTensors.jl:8]: Yeah, I was thinking that too. I think it doesn't even work for symmetric fourth order tensors as well... Might remove it but it is sliightly nice when you want to prescribe a epsilon_11 component for example.

**Comment by KristofferC** [comment]
They generate identical code... But otimes for one case is inlined in my test function

File: src/ContMechTensors.jl (hunk 2 above)
Comment (fredrikekre) [reply src/ContMechTensors.jl:8]:
``` jl
ϵ11 = 1.0
f = (i,j) -> i == j == 1 ? ϵ11 : 0.0
ϵ = SymmetriTensor{2,3}(f)
```

:D

Comment (fredrikekre) [reply src/ContMechTensors.jl:8]: We should at least add tests for it later

Comment (KristofferC) [reply src/ContMechTensors.jl:8]: Yeah, that works pretty good. Hm, maybe just go with that, remove `setindex` and explain in the docs that that is the preferred way.

**Comment by KristofferC** [comment]
Updated: https://gist.github.com/KristofferC/949adf01633dee6a6837a7520de472f5

Looking pretty good now.

**Comment by KristofferC** [comment]
I think we can merge this when tests pass and then I can make a PR to remove setindex and update docs.

**Comment by KristofferC** [comment]
This is with -O3 comparing to non -O3 with static arrays: https://gist.github.com/KristofferC/8c8d39a1f6ecc8a91d85d5d847dc797c

**Comment by KristofferC** [comment]
3 times faster double contraction with 4 order : 2 order. Pretty good.

**Comment by fredrikekre** [comment]
whats up with `dev` and `skew`? They become much faster for `Float32` but slower for `Float64`

**Comment by KristofferC** [comment]
I think it is because I am dividing by 3 instead of multiplying with 1/3. Integer division is more expensive. Your idea with `T(1/3)` fails for integer tensors where `Int(1/3)` gives an inexacterror.

**Comment by fredrikekre** [comment]
What if we divide with `T(3)`?

**Pull request merged**